internal/ccadb/ccadb.go

```diff
@@ -76,104 +62,65 @@ func main() {
 	r.FieldsPerRecord = -1
 	// The Go CSV parser correctly handles quoted fields with internal newlines by default.
 
-	// 3. Read the header row
-	header, err := r.Read()
+	// 3. Verify that necessary headers are present
+	headers, err := r.Read()
 	if err != nil {
-		klog.Exitf("Error reading header row: %v", err)
+		return nil, fmt.Errorf("error reading header row: %v", err)
 	}
 
-	// 4. Dynamically find the required column indices
 	indices := make(map[string]int)
-	requiredHeaders := []string{colSubject, colIssuer, colPEM, colSHA, colUsecase}
-
-	for i, colName := range header {
+	for i, h := range headers {
 		// Clean up the header name (trim potential whitespace or encoding artifacts)
-		cleanName := strings.TrimSpace(colName)
+		cleanName := strings.TrimSpace(h)
 		indices[cleanName] = i
 	}
 
+	allHeadersMap := make(map[string]struct{})
+	for _, h := range KnownHeaders {
+		allHeadersMap[h] = struct{}{}
+	}
+	for _, h := range fetchHeaders {
+		allHeadersMap[h] = struct{}{}
+	}
+
 	minNumColumns := 0
-	// Verify all required headers were found
-	for _, req := range requiredHeaders {
-		i, found := indices[req]
+	for h := range allHeadersMap {
+		i, found := indices[h]
 		if !found {
-			klog.Exitf("Required column not found in CSV header: %s", req)
+			return nil, fmt.Errorf("required column %q not found in CSV headers %q", h, headers)
 		}
 		if i+1 > minNumColumns {
 			minNumColumns = i + 1
 		}
 	}
 
-	// 5. Set up the output file
-	outFile, err := createFile(*outputFilename)
-	if err != nil {
-		klog.Exitf("Error creating output file: %v", err)
-	}
-
-	defer func() {
-		if err := outFile.Close(); err != nil {
-			klog.Errorf("Error closing %q: %v", outFile.Name(), err)
-		}
-	}()
-
-	// 6. Process the remaining records
+	// 4. Process records
+	rows := [][][]byte{}
 	for {
 		row, err := r.Read()
 		if err == io.EOF {
 			break // End of file
 		}
 		if err != nil {
-			klog.Exitf("Malformed record: %v", err)
+			return nil, fmt.Errorf("malformed record: %v", err)
 		}
 
 		// Ensure the record is long enough before attempting to access fields
 		if len(row) < minNumColumns {
-			klog.Exitf("Row is too short: %q", row)
+			return nil, fmt.Errorf("row is too short: %q", row)
 		}
 
-		usecase := row[indices[colUsecase]]
-		if !strings.Contains(usecase, usecaseServerAuth) {
+		usecase := row[indices[ColUsecase]]
+		if !strings.Contains(usecase, UsecaseServerAuth) {
```

Comment:
Is it worth being lenient wrt to capitals/lower case here?

Reply:
Done!